src/base/net/downloadmanager.cpp

```diff
@@ -210,6 +212,15 @@ bool Net::DownloadManager::deleteCookie(const QNetworkCookie &cookie)
     return static_cast<NetworkCookieJar *>(m_networkManager.cookieJar())->deleteCookie(cookie);
 }
 
+bool Net::DownloadManager::hasSupportedScheme(const QString &url)
+{
+    const QStringList schemes = instance()->m_networkManager.supportedSchemes();
```

Comment:
Shouldn't this local var be static and not re-initialized in each function call?

Reply:
Having it static makes it takes permanent memory even when rarely used, I don't prefer it.

Reply:
>even when rarely used

I guess you're right. I wrote the suggestion before reading the rest of your substitutions. Indeed this function isn't called often enough to incur noticeable slowdown.